### PR TITLE
Update dockerfiles for osbs

### PR DESCRIPTION
**Pull request by kevinrizza**
-Use openshift ci registry for base images
-Add downstream ocp dockerfile

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,10 +1,10 @@
 
-FROM openshift/origin-release:golang-1.10 as builder
+FROM registry.svc.ci.openshift.org/ocp/builder:golang-1.10 AS builder
```
Comment (aravindhp) [inline Dockerfile]: Is VPN access required for this? We need ensure this works if anyone tried using the operator. Moreover I see that the OLM [Dockerfile](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/Dockerfile#L1) does not do this and they have gone through the OSBS process.

File: tmp/build/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM openshift/origin-base
+FROM registry.svc.ci.openshift.org/openshift/origin-v4.0:base
```
Comment (aravindhp) [inline tmp/build/Dockerfile]: Same question here

File: Dockerfile.rhel7 [hunk 3]
```diff
@@ -0,0 +1,18 @@
+
```
Comment (aravindhp) [inline Dockerfile.rhel7]: If this is this identical to the Dockerfile won't a symbolic link suffice?

Comment (kevinrizza) [reply Dockerfile.rhel7]: They are not identical, the base images are different.

File: Dockerfile (hunk 1 above)
Comment (kevinrizza) [reply Dockerfile]: I did just test this and it does work without VPN. Moreover, I believe this is a standard that the release team is trying to enforce. Take a look at this documentation for more detail:

https://mojo.redhat.com/docs/DOC-1179058#jive_content_id_Request_Addition_to_ART_Build_System

There are several operator repos that are using these versions of the base image. It seems likely that the OLM source code is just not following the standard.

Comment (kevinrizza) [reply Dockerfile]: For example:

https://github.com/openshift/cluster-kube-apiserver-operator/blob/master/Dockerfile.rhel7#L1

https://github.com/openshift/cluster-samples-operator/blob/master/Dockerfile#L1

https://github.com/openshift/cluster-openshift-apiserver-operator/blob/master/Dockerfile.rhel7#L1

https://github.com/openshift/cluster-ingress-operator/blob/master/Dockerfile.rhel7#L1

**Comment by kevinrizza** [comment]
@tkashem @aravindhp Merged to master due to conflicts, so this needs to be re-approved (no actual changes, just a file path change for the sdk dockerfile). Can someone double check?

**Comment by aravindhp** [comment]
Why are we seeing a merge commit? Did you rebase correctly?

**Comment by kevinrizza** [comment]
> Why are we seeing a merge commit? Did you rebase correctly?
 
Must not have rebased properly last time, I just did it again and there is only one commit. Should be good now.

**Pull request merged**